module/VuFind/tests/unit-tests/src/VuFindTest/Solr/WriterTest.php

```diff
@@ -53,10 +55,13 @@ class WriterTest extends \PHPUnit\Framework\TestCase
      */
     public function testCommit()
     {
-        $bm = $this->getBackendManagerWithMockSolr();
+        $client = $this->getMockBuilder(\Laminas\Http\Client::class)
```

Comment:
...or, given that this logic is used in multiple tests, maybe it could be part of a trait for testing with HTTP clients. But again, maybe that's overkill.

Reply:
I considered a generic getter, but I think it would be a too far-fetched expectation that setOptions would always be called with just the timeout and just once, and if you start adding more options, the helper method easily becomes harder to use and understand than just wiring things up case by case.

I managed to convince myself to add `getMockHttpClientExpectingOnlySetTimeout` locally, however. Coming up.

Reply:
Makes sense!